apps/btshell/src/main.c

```diff
@@ -1074,6 +1134,11 @@ btshell_gap_event(struct ble_gap_event *event, void *arg)
         console_printf("advertise complete; reason=%d, instance=%u, handle=%d\n",
                        event->adv_complete.reason, event->adv_complete.instance,
                        event->adv_complete.conn_handle);
+
+        if (ext_adv_restart[event->adv_complete.instance].restart) {
```

Comment:
I think you should either skip `if` here (so we always have proper `conn_handle` set in array for debugging) or assign invalid handle in `else`, just to have some consistent state here